relays/generic/exchange/src/exchange_loop.rs

```diff
@@ -85,50 +85,32 @@ pub async fn run<P: TransactionProofPipeline>(
 	target_client: impl TargetClient<P>,
 	metrics_params: Option<MetricsParams>,
 	exit_signal: impl Future<Output = ()>,
-) {
+) -> Result<(), String> {
 	let exit_signal = exit_signal.shared();
-	let metrics_global = GlobalMetrics::default();
-	let metrics_exch = ExchangeLoopMetrics::default();
-	let metrics_enabled = metrics_params.is_some();
-	metrics_start(
-		format!("{}_to_{}_Exchange", P::SOURCE_NAME, P::TARGET_NAME),
-		metrics_params,
-		&metrics_global,
-		&metrics_exch,
-	);
-
-	relay_utils::relay_loop::run(
-		relay_utils::relay_loop::RECONNECT_DELAY,
-		source_client,
-		target_client,
-		|source_client, target_client| {
+
+	relay_utils::relay_loop(source_client, target_client)
+		.with_metrics(format!("{}_to_{}_Exchange", P::SOURCE_NAME, P::TARGET_NAME))
+		.loop_metric(ExchangeLoopMetrics::default())?
+		.standalone_metric(GlobalMetrics::default())?
+		.expose(metrics_params)
+		.await?
```

Comment:
I'd prefer to have additional tabs here and in other loop files, but fmt prevents me from doing this :/